source/change-streams/tests/unified/change-streams-resume-allowlist.yml

```
@@ -0,0 +1,1421 @@
+schemaVersion: "1.7"
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+      ignoreCommandMonitoringEvents: [ killCursors ]
+      useMultipleMongoses: false
+  - client:
+      id: &globalClient globalClient
+      useMultipleMongoses: false
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: *database0
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: *collection0
+  - database:
+      id: &globalDatabase0 globalDatabase0
+      client: *globalClient
+      databaseName: *database0
+  - collection:
+      id: &globalCollection0 globalCollection0
+      database: *globalDatabase0
+      collectionName: *collection0
+
+tests:
+  - description: change stream resumes after a network error
+    runOnRequirements:
+      - minServerVersion: "4.2"
+        topologies:
+          - replicaset
+          - sharded
+          - load-balanced
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: globalClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode:
+              times: 1
+            data:
+              failCommands:
+                - getMore
+              closeConnection: true
+      - name: createChangeStream
+        object: *collection0
+        arguments:
+          pipeline: []
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: insertOne
+        object: *globalCollection0
+        arguments:
+          document:
+            x: 1
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult:
+          _id:
+            $$exists: true
+          documentKey:
+            $$exists: true
+          operationType: insert
+          ns:
+            db: *database0
+            coll: *collection0
+          fullDocument:
+            x:
+              $numberInt: "1"
+            _id:
+              $$exists: true
+    expectEvents:
+      - client: *client0
+        ignoreExtraEvents: true
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0
+                cursor: {}
+                pipeline:
+                  - $changeStream: {}
+              commandName: aggregate
+              databaseName: *database0
+          - commandStartedEvent:
+              command:
+                getMore:
+                  $$exists: true
+                collection: *collection0
+              commandName: getMore
+              databaseName: *database0
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0
+                cursor: {}
+                pipeline:
+                  - $changeStream:
+                      resumeAfter:
+                        $$unsetOrMatches:
+                          $$exists:
+                            true
```

This may be more readable:

```suggestion
                  - $changeStream:
                      resumeAfter: { $$unsetOrMatches: { $$exists: true }}
```

Prior art: https://github.com/mongodb/specifications/blob/master/source/unified-test-format/tests/valid-pass/poc-change-streams.yml#L214-L217

---

Done.